### DISCUSSION TITLE
Logikgatter

### DISCUSSION THREAD

**Discussion by Wildeswiesel** [Wildeswiesel/Space-Waffel#6] [General]
Als erstes haben wir mit Prüfungsfragen angefangen, welche auch schnell bearbeitet waren
Eine Aufgabe war z.B.:

```
i=0
while i<5:
  print(i,end=" ")
  if i == 3:
    break
  i += 1
else:
  print("done")
```

Danach haben wir if-else nochmals kennen gelernt und das es manchmal sinnvoller ist andere Methoden zu verwenden
if-Else:
```
def get_grade(score):
  if score >= 60:
    return "D"
  elif score >= 70:
    return "C"
  elif score >= 80:
    return "B"
  elif score >= 90:
    return "A"
  else:
    return "F"

print(get_grade(85))
```
hierbei wird immer D ausgegeben sobald der Wert über 60 liegt, dass heißt hierbei ist auf die Reihenfolge zu achten

als Dictionary

```
# As Dictionary, only one of the statements can be true at a time otherwise there is a "unique key" problem.
def get_grade(score):
  return {
    90 <= score <= 100: "A",
    80 <= score < 90: "B",
    70 <= score < 80: "C",
    60 <= score < 70: "D",
    0 <= score < 60: "F"  # I added the condition 0 because otherwise negative inputs won't return "Invalid score"
  }.get(True, "Invalid score")

print(get_grade(85))
```

Hierbei muss der Wertebereich selbst geschrieben werden und es muss darauf geachtet werden das man beim nächsten einmal < nimmt und nicht <=

Dannach haben wir mit Logikgattern weiter gemacht und haben diese kennengelernt:

AND
OR
NOT
NOR
NAND
XOR
XNOR

Diese Logikgattern konnte man mit den verschiedenen schlüssel begriffe verwenden z.b.:

```
def AND(a,b):
  if a == 1 and b == 1:
    return 1
  else:
    return 0

```

allerdings kann man es auch wie folgt schreiben:

```
print(NAND(1,1))
print(XOR(0,1))
```

Zwischen drin haben wir noch eine kleine übung gemacht.
Ergebniss solte sein:

1
22
333
4444
55555

Meine Lösung war wie folgt:

```
for i in range(1, 6):
    print(str(i) * i)
```

Zu letzt haben wir uns noch mit flip-flops auseinander gesetzt die wie folgt funktionieren:

![image](https://github.com/Wildeswiesel/Space-Waffel/assets/152853691/3195ac45-4ae1-442c-9e2b-410a3a56315d)

**Comment by d4c4n** [comment]
Sehr nice. Würde dir noch empfehlen, hier und wieder auf die tatsächlichen Source-Files in GitHub zu verlinken, weil Leute dann einfach den Code lesen können, Verbesserungen vorschlagen können oder auch einfach den Code klonen können. Aber weiter so!

**Reply by Wildeswiesel** [reply]
alles klar werde ich in meinem nächsten berücksichtigen

**Comment by SvenSrc** [comment]
Good summary of our past lecture
